src/test/rustdoc/cross-crate-hidden-assoc-trait-items.rs

```diff
@@ -0,0 +1,13 @@
+// Regression test for issue #95717: Hide cross-crate doc(hidden) associated items in trait impls.
+
+#![crate_name = "dependent"]
+// edition:2021
+// aux-crate:dependency=cross-crate-hidden-assoc-trait-items.rs
+
+// @has dependent/struct.Ty.html
+// @!has - '//*[@id="associatedtype.Assoc"]' 'type Assoc = ()'
+// @!has - '//*[@id="associatedconstant.ASSOC"]' 'const ASSOC: ()'
+// @has dependent/trait.Tr.html
+// @!has - '//*[@id="associatedtype.Assoc"]' 'type Assoc = ()'
+// @!has - '//*[@id="associatedconstant.ASSOC"]' 'const ASSOC: ()'
```

Comment:
Two things on this test:

1. Please add both an associated type and an associated const which are not doc hidden.
2. Use `matches` to check the whole block, not just that they're not present.

Providing some explanations on why I ask you this: in case we change the formatting of the IDs for associated items, this test will still pass but the items might be displayed. It's because `@!has` isn't that great and which is why we added `@matches` for such cases.